app/dao/services_dao.py

```diff
@@ -429,8 +429,12 @@ def dao_fetch_stats_for_service(service_id, limit_days):
 
 
 def dao_fetch_todays_stats_for_service(service_id):
+    today = date.today()
+    start_date = get_london_midnight_in_utc(today)
+    end_date = get_london_midnight_in_utc(today + timedelta(days=1))
```

Comment:
Interested in whether we want
`get_london_midnight_in_utc(today + timedelta(days=1))`

or whether we want
`end_date = start_date + timedelta(days=1)`

Keen to hear if there is a particular reason you chose the one you did?

Reply:
I used the same strategy used in other queries for consistency.

Reply:
@idavidmcdonald if you did start + 1 day, you'd end up with inconsistent stats when spanning over DST changeover night.

Reply:
```
def test_fetch_stats_for_today_only_includes_today_in_bst(notify_db_session):
    template = create_template(service=create_service())
    # two created email, one failed email, and one created sms
    with freeze_time('2001-03-27T23:59:00'):
        # just before midnight yesterday in UTC
        create_notification(template=template, to_field='1', status='delivered')

    with freeze_time('2001-03-27T00:01:00'):
        # just after midnight yesterday in UTC
        create_notification(template=template, to_field='2', status='failed')


    with freeze_time('2001-03-28T12:00:00'):
        # right_now, we have entered BST at this point but had not for the previous two notifications
        create_notification(template=template, to_field='3', status='created')

        stats = dao_fetch_todays_stats_for_service(template.service_id)

    stats = {row.status: row.count for row in stats}
    assert 'delivered' not in stats
    assert stats['failed'] == 1
    assert stats['created'] == 1
```

Am I right in thinking the fact that this test fails with the code in this PR means that there is a bug with what is written?

We should not expect the notification that is 'delivered' to be found as it was sent yesterday but it is getting picked up.

Or maybe I'm just getting myself all confused with BST fun...

Reply:
Your first notification in the test is created on March 27, you'd need to create it at 2001-03-27 23:00 or later

Reply:
Ha oops, just realised all the dates are in 2001. It's 2021 where the crossover was on March the 28th. It was on a different day in 2001, not the 28th.